src/services/ghost/GhostJetbrainsBridge.ts

```diff
@@ -0,0 +1,371 @@
+		const languageInfo = languageForFilepath(uri)
+		const languageName = languageInfo.name.toLowerCase()
+
+		// Map language names to VSCode language IDs
+		const languageIdMap: { [key: string]: string } = {
+			typescript: "typescript",
+			javascript: "javascript",
+			python: "python",
+			java: "java",
+			"c++": "cpp",
+			"c#": "csharp",
+			c: "c",
+			scala: "scala",
+			go: "go",
+			rust: "rust",
+			haskell: "haskell",
+			php: "php",
+			ruby: "ruby",
+			"ruby on rails": "ruby",
+			swift: "swift",
+			kotlin: "kotlin",
+			clojure: "clojure",
+			julia: "julia",
+			"f#": "fsharp",
+			r: "r",
+			dart: "dart",
+			solidity: "solidity",
+			yaml: "yaml",
+			json: "json",
+			markdown: "markdown",
+			lua: "lua",
+		}
+
+		return languageIdMap[languageName] || languageName
+	}
+
+	/**
+	 * Parse and validate the RPC arguments using Zod schemas
+	 */
+	private parseAndValidateArgs(...args: any[]): DocumentParams {
+		// RPC passes all arguments as a single array in args[0]
+		const argsArray = Array.isArray(args[0]) ? args[0] : args
+
+		// Parse with Zod schema
+		const parsed = InlineCompletionArgsSchema.parse(argsArray)
+		const [documentUri, position, fileContent, languageId, requestId] = parsed
+
+		// Safely extract and normalize parameters
+		const uri = typeof documentUri === "string" ? documentUri : String(documentUri)
+		const pos =
+			typeof position === "object" && position !== null && "line" in position && "character" in position
+				? { line: position.line, character: position.character }
+				: { line: 0, character: 0 }
+		const content = typeof fileContent === "string" ? fileContent : String(fileContent)
+		const langId = typeof languageId === "string" ? languageId : String(languageId || "")
+		const reqId = typeof requestId === "string" ? requestId : String(requestId || "")
+
+		return {
+			uri,
+			position: pos,
+			content,
+			languageId: langId,
+			requestId: reqId,
+		}
+	}
+
+	/**
+	 * Normalize content line endings to LF for consistent processing
+	 * JetBrains may send content with different line endings
+	 */
+	private normalizeContent(content: string): NormalizedContent {
+		const normalizedContent = content.replace(/\r\n/g, "\n").replace(/\r/g, "\n")
+		const lines = normalizedContent.split("\n")
+
+		return {
+			normalizedContent,
+			lines,
+		}
+	}
+
+	/**
+	 * Create a mock VSCode TextDocument from the provided parameters
+	 */
+	private createMockDocument(
+		uri: string,
+		normalizedContent: string,
+		lines: string[],
+		language: string,
+	): vscode.TextDocument {
+		const mockDocument = {
+			uri: vscode.Uri.parse(uri),
+			fileName: uri,
+			isUntitled: false, // Set to false to match real file behavior
+			languageId: language,
+			version: 1,
+			isDirty: false,
+			isClosed: false,
+			eol: vscode.EndOfLine.LF,
+			lineCount: lines.length,
+			save: async () => false,
+			getText: (range?: vscode.Range) => {
```

Comment:
We do have something kinda like this here: `src/services/mocking/MockTextDocument.ts`

I wonder if we could combine them or update that one to work as needed here? Not a huge deal